public-api/src/main/java/org/pmiops/workbench/service/CdrVersionService.java

```diff
@@ -0,0 +1,31 @@
+package org.pmiops.workbench.service;
+
+import org.pmiops.workbench.cdr.CdrVersionContext;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Autowired;
+import javax.inject.Provider;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.pmiops.workbench.db.model.CdrVersion;
+import org.pmiops.workbench.db.dao.CdrVersionDao;
+
+@Service
+public class CdrVersionService {
+    private CdrVersionDao cdrVersionDao;
+
+    @Autowired
+    public CdrVersionService(CdrVersionDao cdrVersionDao) {
+        this.cdrVersionDao = cdrVersionDao;
```

Comment:
I would inject the CdrVersionMapper in the service and remove it from your Controllers. That way you can map to the client object from the service methods.

Reply:
Done thanks.